src/common/PhysicsController.cpp

```diff
@@ -0,0 +1,55 @@
+#include <vector>
+#include <memory>
+#include <iostream>
+
+#include "PhysicsController.hpp"
+
+#include "../common/util.hpp"
+
+PhysicsController::PhysicsController() {}
+
+std::shared_ptr<CircleRigidBody> PhysicsController::createCRB(float radius, sf::Vector2f position) {
+    std::shared_ptr<CircleRigidBody> crb = std::shared_ptr<CircleRigidBody>(
+        new CircleRigidBody(radius, position));
+    mCircleRigidBodies.push_back(crb);
+    return crb;
+}
+
+void PhysicsController::step() {
+    for (auto crb : mCircleRigidBodies) {
+        crb->step();
+    }
+}
+
+/**
+    Handles collision between CircleRigidBodies by pairs.
+    To handle the collision between circle A and B, circle B is moved in the direction
+    of the vector spanning from the center of cirlce A to the center of circle B by the
+    amount needed such that the circles are touching but not intersecting.
+*/
+void PhysicsController::handleCollision() {
+    std::vector<std::shared_ptr<CircleRigidBody>> active_circles;
+    std::copy_if(
```

Comment:
im curious if this act of copying is actually more efficient. copying is super expensive

Reply:
if the goal is to copy these as an optimization, i think it may be premature and not actually beneficial

Reply:
You mean you'd prefer to just check if `circle_a` and `circle_b` are active in the loop below? 
Is that really faster/cleaner? Note that we're copying pointers not entire objects.

Reply:
yes  just checking active state. allocating memory and copying stuff is a lot more expensive than checking a boolean (few asm instructions, vs a call to the kernel to get more memory possibly)

Reply:
anyway since this is the way it is, I think it's fine to leave for now, we'll want to rewrite collision detection with a better data structure eventually anyway so that it's not n^2